### DISCUSSION TITLE
Subscribe to push for all Account calendars

### DISCUSSION THREAD

**Discussion by suhanoves** [ecederstrand/exchangelib#1031] [Q&A] [answered]
Erik, thanks a lot for the great library!
Could you explain to me the moment with `FolderCollection.subscribe_to_push()`

My task is to subscribe to all the Account's calendars _(now I have not found how to get them in less than two requests: I take `account.calendar` and `account.calendar.children`)_.

Perhaps the easiest way to get all subscriptions is to loop through each calendar and call `subscribe_to_push` method. But in this case, the number of requests is equal to the number of calendars.

I found that the `FolderCollection` also has a `subscribe_to_push` method. I thought that by calling it, I would get a tuple of subscriptions with one request, but this method returns a generator, and iterating through it I get only one subscription.

Please tell me how you see the best way to get subscriptions to all the user's calendars?

**Comment by ecederstrand** [comment, marked as answer]
A single subscription can span multiple folders. That's the purpose of `FolderCollection.subscribe_to_push()` - it will create a single subscription covering all folders in the collection.

Something like `FolderCollection(account=account, folders=(account.calendar,) + tuple(account.calendar.walk())).subscribe_to_push()` should do what you want.

You're right that `FolderCollection.subscribe_to_push()` returns a generator. Looking at it again, I think that's an error. I'll have it fixed asap. For now, you can just do `list(FolderCollection.subscribe_to_push())[0]` since the generator should yield only one result.

**Reply by ecederstrand** [reply]
See https://github.com/ecederstrand/exchangelib/commit/3f9274e92630f25b952d6e3870e662ec757cd30d and https://github.com/ecederstrand/exchangelib/commit/7988fa69abac3917ea74d6234bc1ec6c79422413
